core/store/src/trie/shard_tries.rs

```diff
@@ -210,11 +210,13 @@ impl ShardTries {
     }
 
     pub fn update_cache(&self, ops: Vec<(&CryptoHash, Option<&[u8]>)>, shard_uid: ShardUId) {
-        let mut caches = self.0.caches.write().expect(POISONED_LOCK_ERR);
-        let cache = caches
-            .entry(shard_uid)
-            .or_insert_with(|| TrieCache::new(&self.0.trie_config, shard_uid, false))
-            .clone();
+        let cache = {
+            let mut caches = self.0.caches.lock().expect(POISONED_LOCK_ERR);
+             caches
+                .entry(shard_uid)
+                .or_insert_with(|| TrieCache::new(&self.0.trie_config, shard_uid, false))
+                .clone()
```

Comment:
@nagisa do I have to explicitly call `drop(caches)` here? Looks like with this implementation I still get lock contention on `update_cache`

Reply:
No, having it be in a block like this should be good enough. How do you check if this contention is occurring? Are you sure you aren't looking at the very first few calls that would need to do the work of creating the `TrieCache`?

Reply:
@nagisa no I am pretty sure it is legit contention. See the query [here](https://nearinc.grafana.net/explore?schemaVersion=1&panes=%7B%224uh%22:%7B%22datasource%22:%22grafanacloud-traces%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22tempo%22,%22uid%22:%22grafanacloud-traces%22%7D,%22queryType%22:%22traceql%22,%22limit%22:100,%22tableType%22:%22traces%22,%22query%22:%22%7Bresource.service.name%3D%5C%22neard:ed25519:92Dn36qNrgbofdXpXcNDcrHFpRVTK17nzpHWFrRzF5ME%5C%22%20%26%26%20name%3D%5C%22postprocess_block%5C%22%7D%20%3E%3E%20%7Bduration%20%3E%20200ms%7D%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1) and click into the spans to see the time spent on ShardTries::update_cache